src/btree/bt_sync_obsolete.c

```diff
@@ -94,11 +105,41 @@ __sync_obsolete_inmem_evict(WT_SESSION_IMPL *session, WT_REF *ref)
         /* Mark the obsolete page to evict soon. */
         __wt_page_evict_soon(session, ref);
         WT_STAT_CONN_DSRC_INCR(session, checkpoint_cleanup_pages_evict);
+    } else {
+        /*
+         * Limit the number of obsolete time window pages that are marked as dirty to reduce the
+         * load.
+         */
+        if (S2BT(session)->cc_obsolete_tw_pages >=
+          S2C(session)->heuristic_controls.cc_obsolete_tw_pages_dirty)
+            return (0);
+
+        /*
+         * The pages that are completely removed are eliminated once they are obsolete. There is no
+         * point in clearing their on-disk time window information now.
+         */
+        if (WT_TIME_AGGREGATE_HAS_STOP(&newest_ta))
```

Comment:
Can we create a variable so we can use it here and line 87?
```
has_stop = WT_TIME_AGGREGATE_HAS_STOP(&newest_ta);
if (do_visibility_check && has_stop)
...
if(has_stop)
```

Reply:
Done.